### PR TITLE
feat: initial JSON schema

### PR DESCRIPTION
**Pull request by hoffa**
### Issue #, if available

#1133 

### Description of changes

Includes all SAM-specific properties; the values (marked `Unknown`) will be replaced over time.

Adds new command `make schema` to generate the JSON schema from `schema.py`.

All transform and integration test templates are validated against the schema with `make pr`.

### Checklist

- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [ ] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

### Examples?

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ssenchenko — APPROVED** [review]
That's the way to generate schema? Cool!

**Comment by GavinZZ** [comment]
Would it affect error test? E.g. if I want to add a invalid field which is not supported by the schema.

**Comment by hoffa** [comment]
> Would it affect error test? E.g. if I want to add a invalid field which is not supported by the schema.

Nope, check `validate.sh`.

**Review by GavinZZ — APPROVED** [review]

**Comment by GavinZZ** [comment]
> > Would it affect error test? E.g. if I want to add a invalid field which is not supported by the schema.
> 
> Nope, check `validate.sh`.

Nice!

File: Makefile [hunk 1]
```diff
@@ -32,10 +32,15 @@ lint:
 	mypy --strict samtranslator bin
 	# Linter performs static analysis to catch latent bugs
 	pylint --rcfile .pylintrc samtranslator
+	# Ensure templates adhere to JSON schema
+	./validate.sh
```
Comment (aahung) [inline Makefile:36]: move it to `bin/`?

Comment (hoffa) [reply Makefile]: Done

**Review by acristin — APPROVED** [review]
lgtm

**Review by aahung — APPROVED** [review]

**Pull request merged**